CRM/SMS/Form/Provider.php

```diff
@@ -113,8 +113,12 @@ public function setDefaultValues() {
     $name = CRM_Utils_Request::retrieve('key', 'String', $this, FALSE, NULL);
     if ($name) {
       $defaults['name'] = $name;
-      $provider = CRM_SMS_Provider::singleton(['provider' => $name]);
-      $defaults['api_url'] = $provider->_apiURL;
+      try {
```

Comment:
So my hesitation is around this one & the next one which adds handling. The previous behaviour here was a fatal when the provider was invalid. Without the catch here the form layer would give the equivalent of that - but with the catch it seems an obscure edge case bug would be ignored